### PR TITLE
docs(analytics): boolean parameters are not supported for GA custom definitions.

### PR DESCRIPTION
**Pull request by saveKenny**
closes https://github.com/FirebaseExtended/flutterfire/issues/7025

## Description

From Firebase official support team:
However, since your ??parameters with ?string ?data type as values are shown in the report, it is safe to assume that there's no issue in your implementation. To give some insights regarding custom dimensions and metrics, do note that custom dimensions refer to text-type parameters, while metrics refer to number-type parameters. So before logging these parameters, you have to register accordingly the ?string type parameters as custom dimensions and ?integer ?(double/long) type parameters as custom metrics in the Custom Definitions page for them to appear in the report. As for your boolean type parameters, you can register them as custom dimensions. It will depend on your use case and how you indicate your boolean values (e.g. "True" or "False", "Yes" or "No"). You can check this article to learn more.

### CODE REVIEW

**Comment by saveKenny** [comment]
@googlebot I signed it!

File: packages/firebase_analytics/firebase_analytics/example/lib/main.dart [hunk 1]
```diff
@@ -72,7 +72,7 @@ class _MyHomePageState extends State<MyHomePage> {
         'int': 42,
         'long': 12345678910,
         'double': 42.0,
-        'bool': true,
+        'bool': true.toString(),
```
Comment (russellwheatley) [inline packages/firebase_analytics/firebase_analytics/example/lib/main.dart:75]: Do you mind making a note that only numbers and strings are accepted (hence why stringified)? Also paste in this link perhaps: https://developers.google.com/analytics/devguides/collection/analyticsjs/custom-dims-mets#overview

**Review by russellwheatley — CHANGES_REQUESTED** [review]

File: packages/firebase_analytics/firebase_analytics/example/lib/main.dart (hunk 1 above)
Comment (russellwheatley) [reply packages/firebase_analytics/firebase_analytics/example/lib/main.dart:75]: also point to master branch please.

Comment (saveKenny) [reply packages/firebase_analytics/firebase_analytics/example/lib/main.dart:75]: > Do you mind making a note that only numbers and strings are accepted (hence why stringified)? Also paste in this link perhaps: https://developers.google.com/analytics/devguides/collection/analyticsjs/custom-dims-mets#overview

What do you mean about _making a note_?

Comment (russellwheatley) [reply packages/firebase_analytics/firebase_analytics/example/lib/main.dart:75]: a comment: 
```dart
    'int': 42,
        'long': 12345678910,
        'double': 42.0,
        'bool': true,
		// comment in code
        'bool': true.toString(),

```

Comment (saveKenny) [reply packages/firebase_analytics/firebase_analytics/example/lib/main.dart:75]: > also point to master branch please.

I'm new in github, do I need to open a different request or I can point to the master from this one?

Comment (russellwheatley) [reply packages/firebase_analytics/firebase_analytics/example/lib/main.dart:75]: point to master on this PR.

Comment (saveKenny) [reply packages/firebase_analytics/firebase_analytics/example/lib/main.dart]: > point to master on this PR.

How do I do this from this PR?

File: packages/firebase_analytics/firebase_analytics/example/lib/main.dart [hunk 2]
```diff
@@ -72,7 +72,7 @@ class _MyHomePageState extends State<MyHomePage> {
         'int': 42,
         'long': 12345678910,
         'double': 42.0,
-        'bool': true,
+        'bool': true.toString(), // Only strings and integers are supported for GA custom event parameters https://developers.google.com/analytics/devguides/collection/analyticsjs/custom-dims-mets#overview
```
Comment (russellwheatley) [inline packages/firebase_analytics/firebase_analytics/example/lib/main.dart:75]: Well done on figuring out the branch change 😄. Do you mind changing the wording and formatting to below noted?  
```dart
		'int': 42,
        'long': 12345678910,
        'double': 42.0,
		// Only strings and numbers (ints & doubles) are supported for GA custom event parameters:
		// https://developers.google.com/analytics/devguides/collection/analyticsjs/custom-dims-mets#overview
        'bool': true,
```

**Review by russellwheatley — CHANGES_REQUESTED** [review]

**Review by Salakar — APPROVED** [review]

**Review by russellwheatley — APPROVED** [review]
LGTM

**Pull request merged**